common/k8s/src/kube_stats/container_stats.rs

```diff
@@ -0,0 +1,437 @@
+        c_status: &'a ContainerStatus,
+        c_state: &'a ContainerState,
+        raw_cpu_usage: &'a str,
+        raw_memory_usage: &'a str,
+    ) -> ContainerStatsBuilder<'a> {
+        ContainerStatsBuilder {
+            c,
+            c_status,
+            c_state,
+            raw_cpu_usage,
+            raw_memory_usage,
+        }
+    }
+}
+
+pub struct ContainerStatsBuilder<'a> {
+    c: &'a Container,
+    c_status: &'a ContainerStatus,
+    c_state: &'a ContainerState,
+    raw_cpu_usage: &'a str,
+    raw_memory_usage: &'a str,
+}
+
+impl ContainerStatsBuilder<'_> {
+    pub fn new<'a>(
+        c: &'a Container,
+        c_status: &'a ContainerStatus,
+        c_state: &'a ContainerState,
+        raw_cpu_usage: &'a str,
+        raw_memory_usage: &'a str,
+    ) -> ContainerStatsBuilder<'a> {
+        ContainerStatsBuilder {
+            c,
+            c_status,
+            c_state,
+            raw_cpu_usage,
+            raw_memory_usage,
+        }
+    }
+
+    pub fn build(self) -> ContainerStats {
+        let container = self.c.name.clone();
+
+        let memory_usage = convert_memory_usage_to_bytes(self.raw_memory_usage);
+        let cpu_usage = convert_cpu_usage_to_milli(self.raw_cpu_usage);
+
+        let mut image = String::new();
+        let mut image_tag = String::new();
+        let state;
+        let mut last_state = String::new();
+        let mut last_reason = String::new();
+
+        let mut cpu_limit = None;
+        let mut cpu_request = None;
+        let mut memory_limit = None;
+        let mut memory_request = None;
+        let mut last_started = None;
+        let mut last_finished = None;
+
+        let mut container_age: i64 = 0;
+        let mut started: i64 = 0;
+
+        let restarts = self.c_status.restart_count;
+        let ready = self.c_status.ready;
+
+        if self.c.image.is_some() {
+            let container_image = self.c.image.clone().unwrap();
+
+            let split = container_image.split_once(':');
+
+            if let Some(..) = split {
+                image = split.unwrap().0.to_string();
+                image_tag = split.unwrap().1.to_string();
+            }
+        }
+
+        let running = self.c_state.running.as_ref();
+        let terminated = self.c_state.terminated.as_ref();
+
+        if let Some(..) = running {
+            state = "Running".to_string();
+
+            let started_at = running.unwrap().started_at.as_ref().map(|s| s.0);
+
+            if let Some(..) = started_at {
+                container_age = Local::now()
+                    .signed_duration_since(started_at.unwrap())
+                    .num_milliseconds();
+
+                started = started_at.unwrap().timestamp_millis();
+            }
+        } else if terminated.is_some() {
+            state = "Terminated".to_string()
+        } else {
+            state = "Waiting".to_string()
+        }
+
+        let last_status_state = self.c_status.last_state.as_ref();
```

Comment:
Can we use something like the following to avoid the unwraps on line 150, etc?

```rust
if let Some(last_status_state) = self.c_status.last_state.as_ref() {

}
```

Reply:
Same for all the other `if let Some(..)` blocks

Reply:
cleaned up a bunch of these